paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -216,72 +438,305 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+                           const paddle::framework::Tensor* input,
+                           const paddle::framework::Tensor* filter,
+                           const paddle::framework::Tensor* bias,
+                           paddle::framework::Tensor* output,
+                           ConvInfo* convinfo, MkldnnInfo* mkldnninfo) const {
+    const T* input_data = input->data<T>();
+    const K* filter_data = filter->data<K>();
+    bool is_INT8 = true;
+    auto scale_in_data = ctx.Attr<float>("Scale_in");
+    auto scale_in_eltwise_data = ctx.Attr<float>("Scale_in_eltwise");
+    auto scale_weights_data = ctx.Attr<std::vector<float>>("Scale_weights");
+    auto scale_out_data =
+        mkldnninfo->force_fp32_output ? 1.0f : ctx.Attr<float>("Scale_out");
+
+    bool is_multi_channel = scale_weights_data.size() > 1 ? true : false;
+
+    std::vector<float> output_shift_scale;
+    float sum_scale = 1.0f;
+    int count =
+        is_multi_channel
+            ? (convinfo->g > 1
+                   ? (*convinfo->weights_tz)[1] * (*convinfo->weights_tz)[0]
+                   : (*convinfo->weights_tz)[0])
+            : 1;
+    output_shift_scale.resize(count);
+#pragma omp parallel for if (count > 1)
+    for (int i = 0; i < count; i++) {
+      if (scale_weights_data[i] == 0.0)
+        output_shift_scale[i] = scale_out_data;  // weights data will contain 0
+                                                 // in some models, then weights
+                                                 // scale couldn't be calculated
+      else
+        output_shift_scale[i] =
+            scale_out_data / (scale_in_data * scale_weights_data[i]);
+    }
+    if (mkldnninfo->fuse_residual_conn) {
+      sum_scale = scale_out_data / scale_in_eltwise_data;
+    }
+
+    auto user_src_md = platform::MKLDNNMemDesc(
+        {*convinfo->src_tz}, paddle::framework::ToMKLDNNDataType(input->type()),
+        input->format());
+    auto user_weights_md = platform::MKLDNNMemDesc(
+        {*convinfo->weights_tz}, platform::MKLDNNGetDataType<K>(),
+        ((convinfo->g) == 1) ? mkldnn::memory::format::oihw
+                             : mkldnn::memory::format::goihw);
+
+    /* create memory descriptor for convolution without specified format
+     * ('any') which lets a primitive (convolution in this case) choose
+     * the memory format preferred for best performance
+    */
+    std::string data_format = ctx.Attr<std::string>("data_format");
+    auto chosen_memory_format =
+        platform::data_format_to_memory_format(data_format);
+
+    auto bias_tz = paddle::framework::vectorize2int(bias->dims());
+
+    auto src_md = platform::MKLDNNMemDesc(
+        *convinfo->src_tz, memory::data_type::u8, chosen_memory_format);
+    auto weights_md = platform::MKLDNNMemDesc(
+        *convinfo->weights_tz, memory::data_type::s8, chosen_memory_format);
+
+    auto dst_dt = mkldnninfo->fuse_relu
```

Comment:
The choice of `dst_dt` type could be turned into a `switch` statement which would consider `fuse_relu`, `force_fp32_output` and `fuse_residual_conn` which would make it a bit more readable.